2020/dec17/energysource/source3d/state.go

```diff
@@ -0,0 +1,83 @@
+package source3d
+
+import (
+	"fmt"
+)
+
+// State is an implementation of the Experimental Energy Source
+// It represents the state of the Source at a given point in the iterative dimension (think time, for example)
+type State struct {
+	active map[coord3D]struct{}
+	min    coord3D
+	max    coord3D
+}
+
+type coord3D struct {
+	x, y, z int
+}
+
+func New() *State {
+	return &State{active: make(map[coord3D]struct{})}
+}
+
+// Activate sets the given cell to Active
+func (s *State) Activate(x, y, z int) {
+	c := coord3D{x: x, y: y, z: z}
```

Comment:
it's sad to see this object has already been created right before the call to this func in `iterate()`